components/dot_menu/post_reminder_submenu.tsx

```diff
@@ -0,0 +1,135 @@
+            endTime = currentDate.add(2, 'hours');
+            break;
+        case 'tomorrow':
+            // add one day in current date
+            endTime = currentDate.add(1, 'day');
+            break;
+        }
+
+        dispatch(addPostReminder(props.userId, props.post.id, toUTCUnix(endTime.toDate())));
+    };
+
+    const setCustomPostReminder = (): void => {
+        const postReminderCustomTimePicker = {
+            modalId: ModalIdentifiers.POST_REMINDER_CUSTOM_TIME_PICKER,
+            dialogType: PostReminderCustomTimePicker,
+            dialogProps: {
+                postId: props.post.id,
+                currentDate: new Date(),
+            },
+        };
+        dispatch(openModal(postReminderCustomTimePicker));
+    };
+
+    const postReminderSubMenuItems =
+        postReminderTimes.map(({id, label, labelDefault}) => {
+            const labels = (
+                <FormattedMessage
+                    id={label}
+                    defaultMessage={labelDefault}
+                />
+            );
+            let trailing: JSX.Element | null = null;
+
+            if (id === 'tomorrow') {
+                const tomorrow = getCurrentMomentForTimezone(props.timezone).add(1, 'day').toDate();
+                trailing = (
+                    <span className={`postReminder-${id}_timestamp`}>
+                        <FormattedDate
+                            value={tomorrow}
+                            weekday='short'
+                        />
+                        {', '}
+                        <FormattedTime
+                            value={tomorrow}
+                            timeStyle='short'
+                            hour12={!props.isMilitaryTime}
+                        />
+                    </span>
+                );
+            }
+            return (
+                <Menu.Item
+                    key={`remind_post_options_${id}`}
+                    id={`remind_post_options_${id}`}
+                    labels={labels}
+                    trailingElements={trailing}
+                    onClick={id === 'custom' ? () => setCustomPostReminder() : () => setPostReminder(id)}
+                />
+            );
+        });
+
+    return (
+        <Menu.SubMenu
+            id={`remind_post_${props.post.id}`}
+            labels={
+                <FormattedMessage
+                    id='post_info.post_reminder.menu'
+                    defaultMessage='Remind'
+                />
+            }
+            leadingElement={<ClockOutlineIcon size={18}/>}
+            trailingElements={<ChevronRightIcon size={16}/>}
+            menuId={`remind_post_${props.post.id}-menu`}
+            forceOpenOnLeft={true}
+        >
+            <h5 className={'postReminderMenuHeader'}>
+                {formatMessage(
+                    {id: 'post_info.post_reminder.sub_menu.header',
+                        defaultMessage: 'Set a reminder for:'},
+                )}
+            </h5>
```

Comment:
should we make `Menu.Header` component? can be used in other menus as well

Reply:
Seems like one-off case to me